web/src/presence.ts

```diff
@@ -305,6 +326,48 @@ export function last_active_date(user_id: number): Date | undefined {
     return new Date(info.last_active * 1000);
 }
 
+export let fetch_presence_for_user = (user_id: number): void => {
+    const url = `json/users/${user_id}/presence`;
+
+    channel.get({
+        url,
+        success(data: unknown) {
+            const parsed_data = user_last_seen_response_schema.safeParse(data);
+
+            if (parsed_data.success) {
+                const response = parsed_data.data;
+
+                if (response.result === "success" && response.presence) {
+                    const {aggregated} = response.presence;
+                    presence_info.set(user_id, {
+                        status: aggregated.status,
+                        last_active: aggregated.timestamp,
+                    });
+
+                    // Update the user's last seen time in the user card
+                    // popover once we have their presence information.
+                    $(".user-last-seen-time").text(
+                        timerender.last_seen_status_from_date(
+                            new Date(aggregated.timestamp * 1000),
+                        ),
```

Comment:
This selector is too broad -- because network delay has happened, the user could be looking at a different user's card already!